### PR TITLE
[monorepo] fix: add initial codecoverage

### PR DESCRIPTION
**Pull request by Wayonb**
- add initial code coverage for explorer front-end and nemesis generator
- add build for website
- add linting for nem-docker
- CodeQL

### CODE REVIEW

File: nem-docker/boot.sh [hunk 1]
```diff
@@ -13,36 +13,35 @@ docker ps -a | grep mynem_container > /dev/null && docker rm mynem_container
 
 
 # determine which custom configs to mount
-
-set -x
```
Comment (Jaguar0625) [inline nem-docker/boot.sh:17]: why did you delete this ?

File: nem-docker/clean-logs.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-# delet elog files older than the 9th logfile 
+# delete elog files older than the 9th logfile
```
Comment (Jaguar0625) [inline nem-docker/clean-logs.sh]: log

File: website/scripts/ci/setup_lint.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -ex
+
+npm install --legacy-peer-deps
```
Comment (Jaguar0625) [inline website/scripts/ci/setup_lint.sh:5]: why `--legacy-peer-deps` ?

File: website/scripts/ci/test.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -ex
+
+TEST_MODE=$([ "$1" = "code-coverage" ] && echo "test:jenkins" || echo "test")
+# CI=true npm run "${TEST_MODE}"
```
Comment (Jaguar0625) [inline website/scripts/ci/test.sh:6]: commented out because we're not running tests?

File: nem-docker/boot.sh (hunk 1 above)
Comment (Wayonb) [reply nem-docker/boot.sh:17]: dont think debug logs are needed in customer scripts but can add it back.

File: nem-docker/clean-logs.sh (hunk 2 above)
Comment (Wayonb) [reply nem-docker/clean-logs.sh]: ha, I am blind and need to get new glasses. 🤣

File: website/scripts/ci/setup_lint.sh (hunk 3 above)
Comment (Wayonb) [reply website/scripts/ci/setup_lint.sh:5]: it seems with react 17+ and npm 7+ the build will fail due to peer dependencies.  This is due to a lot of third party apps not upgraded.
https://weekendprojects.dev/posts/how-to-use-npm-legacy-peer-deps-command/#the---legacy-peer-deps-flag-and-react

File: website/scripts/ci/test.sh (hunk 4 above)
Comment (Wayonb) [reply website/scripts/ci/test.sh:6]: There is currently no tests to run for the website project so it fails the CI.
They can remove comment  when tests are added

File: nem-docker/boot.sh (hunk 1 above)
Comment (Jaguar0625) [reply nem-docker/boot.sh:17]: would prefer to keep it

**Review by Jaguar0625 — DISMISSED** [review]

File: nem-docker/boot.sh (hunk 1 above)
Comment (Jaguar0625) [reply nem-docker/boot.sh:17]: any reason not to use set -ex ?

Comment (Wayonb) [reply nem-docker/boot.sh:17]: ``set -eu`` is already at the top of the file.

**Review by Jaguar0625 — APPROVED** [review]
